src/test/java/GreetzLoginFavouritesTest.java

```diff
@@ -43,103 +55,79 @@ public boolean isLoggedIn(String firstName) {
-            if (favoiriteButton.getAttribute("class").equals("b-favourite b-favourite_selected")) {
-                favoiriteButton.click();
-                Thread.sleep(3000);
-            }
+        driver.get("https://www.greetz.nl/ballonnen/denken-aan");
+        Thread.sleep(6000);
 
+        List<WebElement> balloons = driver.findElements(By.xpath("//*[@class='b-products-grid__item']"));
+        WebElement chosenBalloon = randomItem(balloons);
+        WebElement favoiriteButton = chosenBalloon.findElement(By.xpath(".//a[2]/div"));
+        if (favoiriteButton.getAttribute("class").equals("b-favourite b-favourite_selected")) {
             favoiriteButton.click();
-
-            String expectedItemTitle = chosenBalloon.findElement(By.xpath(".//div[2]")).getText();
-            String expectedItemPrice = chosenBalloon.findElement(By.xpath(".//div[3]")).getText();
             Thread.sleep(3000);
+        }
 
-            WebElement myGreetz = driver.findElement(By.linkText("MyGreetz"));
-            myGreetz.click();
-            Thread.sleep(3000);
+        favoiriteButton.click();
 
-            WebElement favorieten = driver.findElement(By.xpath("//span[text()='Favorieten']"));
-            favorieten.click();
-            Thread.sleep(3000);
+        String expectedItemTitle = chosenBalloon.findElement(By.xpath(".//div[2]")).getText();
+        String expectedItemPrice = chosenBalloon.findElement(By.xpath(".//div[3]")).getText();
+        Thread.sleep(3000);
 
-            WebElement favItem = driver.findElement(By.xpath("//*[@class='favorite-item']"));
-            favItem.click();
-            Thread.sleep(3000);
+        WebElement myGreetz = driver.findElement(By.linkText("MyGreetz"));
+        myGreetz.click();
+        Thread.sleep(3000);
 
-            String actualItemTitle = driver.findElement(By.xpath("//*[@ng-bind='gift.title']")).getText();
-            String actualItemPrice = driver.findElement(By.xpath("//*[@class='price-normal']")).getText().substring(2);
+        WebElement favorieten = driver.findElement(By.xpath("//span[text()='Favorieten']"));
+        favorieten.click();
+        Thread.sleep(3000);
 
-            Assert.assertEquals(actualItemPrice, expectedItemPrice, "Prices: ");
-            Assert.assertEquals(actualItemTitle, expectedItemTitle, "Names: ");
-        } finally {
-            driver.quit();
-        }
+        WebElement favItem = driver.findElement(By.xpath("//*[@class='favorite-item']"));
+        favItem.click();
+        Thread.sleep(3000);
+
+        String actualItemTitle = driver.findElement(By.xpath("//*[@ng-bind='gift.title']")).getText();
+        String actualItemPrice = driver.findElement(By.xpath("//*[@class='price-normal']")).getText().substring(2);
+
+        Assert.assertEquals(actualItemPrice, expectedItemPrice, "Prices: ");
+        Assert.assertEquals(actualItemTitle, expectedItemTitle, "Names: ");
```

Comment:
what if the name matches but the price not? I would suggest to use `SoftAssert`

Reply:
Addressed in 5811ad0.

Added `SoftAssert`.